contracts/PoolManager.sol

```diff
@@ -201,7 +201,7 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
     }
 
     modifier onlyByLocker() {
-        address locker = lockedBy[lockedBy.length - 1];
+        address locker = locks[lockIndex].locker;
```

Comment:
nit: `locks[lockIndex].locker` can be used directly in the comparison. `locker` will be used twice(to save gas) only when it reverts. So actually, it will cost more gas for succeed cases.

Reply:
nice catch! fixed